USWDS theme for keycloak

awolfe76 opened the pull request:
_The number of files (119) is due to the files in USWDS. We can probably clean it up later, but it doesn't hurt anything._

- adds in [uswds](https://standards.usa.gov/) files
- customizes login
![hmda-login](https://cloud.githubusercontent.com/assets/2932572/20795109/6f4ef730-b79d-11e6-936b-a0766e0ad8ca.png)
- customizes registration
![hmda-register](https://cloud.githubusercontent.com/assets/2932572/20795118/78752fc8-b79d-11e6-8734-c3802a01a30f.png)
- customizes password reset
![hmda-reset-pwd](https://cloud.githubusercontent.com/assets/2932572/20795126/820780f4-b79d-11e6-9c02-c2252e376f24.png)
- custom styles for errors
![hmda-errors](https://cloud.githubusercontent.com/assets/2932572/20795138/8c92d5d2-b79d-11e6-8858-cbaf1d59b2d4.png)

hkeeler commented:
@awolfe76, this is awesome!  Taking a crack at getting this all running locally now.

hkeeler commented on keycloak/themes/hmda/login/register.ftl:
```diff
@@ -155,43 +110,22 @@
-
+                    return '<div class="usa-grid-full">' +
+                           //'    <div styles="vertical-align:middle;" class="col-md-1">' +
+                           //'      <h1>' + getStatusIcon(getFormEmail(), institution.domain[0]) + '</h1>' +
```
Is there a plan to re-incorporate some form of status indicator?  Without this, there's no way for the user to know ahead of time if the institution they've selected is going to work.

I realize there's a bit less real estate now, so this is likely a question of _where do we put it?_

awolfe76 replied:
Yes. But I think it's ok for now, we'll figure it out. That's why I left the reminder there. 😄 
I hope that the meeting today will tell us a bit more.

hkeeler replied:
Cool by me.  I created https://github.com/cfpb/hmda-platform-auth/issues/32 so we don't forget about it.

hkeeler commented:
@awolfe76, just curious, how much did you have to change the templates themselves?  Are these copied directly, and just the CSS classes updated, or was there more too it.  I'm asking primarily to better understand how much work it is going to be with each upgrade, and how we're going to have to re-check the templates for necessary changes each time.

One thought, it may be more upgrade-friendly to keep keycloak variables for the CSS class names, and update the values in the `theme.properties` file(s) as they do in there templates.

Thoughts?

awolfe76 commented on keycloak/themes/hmda/login/theme.properties:
```diff
@@ -1,5 +1,5 @@
-styles=lib/patternfly/css/patternfly.css lib/zocial/zocial.css lib/select2-3.4.1/select2.css css/login.css
+styles=css/uswds.min.css lib/select2-3.4.1/select2.css css/select2-custom.css
 
-institutionSearchUri={{INSTITUTION_SEARCH_URI}}
```
Yeah, sorry. Didn't commit that before but must have slipped through at some point.

awolfe76 commented:
@hkeeler 
> how much did you have to change the templates themselves? 

It was all of the above, copied, modified CSS classes, and modified the HTML structure too (mostly due to the new CSS classes).

> I'm asking primarily to better understand how much work it is going to be with each upgrade, and how we're going to have to re-check the templates for necessary changes each time.

I'm not worried about the upgrade path on the templates at all, even without using `theme.properties` to store the classes. It seems like they did it that way so others could easily import their themes. Unless we decide to make our own base, can't see why we would, or they change their theme engine, totally different story, we'll be fine.

awolfe76 commented:
@hkeeler - when looking at the files after the conversation here I thought it would be good to clean them up a bit, see the last 2 commits.

To view without whitespace changes:
https://github.com/cfpb/hmda-platform-auth/pull/31/commits/d20c723b3f40054f6ed588ad4cf1a08542975c6a?w=1
https://github.com/cfpb/hmda-platform-auth/pull/31/commits/d1252cf196aeded03c592ce1b17ea50fee39d505?w=1

hkeeler commented:
@awolfe76, thanks for the cleanup.  This looks good to me.  I'd like @wpears to weigh in if possible before merging.

Also, I'm working on a _How to Upgrade Keycloak_ doc.  I'll reach out when I get to the bit about templates.
